translations/static/js/base.js

```diff
@@ -112,11 +116,54 @@ function checkDropdownPosition(dropdown, dropdownContent) {
   }
 }
 
-function initializeAll() {
-  initializeTableSorting();
-  initializeSidebarMenu();
-  initializeDropdowns();
+// Convert UTC ISO in data-utc-date to browser timezone
+function formatDates() {  
+  const els = document.querySelectorAll(".utc-date");  
+  
+  if (els.length === 0) {
+    return;
+  }
+  
+  els.forEach(function (el, index) {
+    const iso = el.getAttribute("data-utc-date");    
+    
+    if (!iso || iso === "Never") {
+      return;
+    }
+    
+    try {
+      const d = new Date(iso);
+      if (isNaN(d.getTime())) {
+        return;
+      }
+      
+      const formatted = new Intl.DateTimeFormat(navigator.language, {
+        year: "numeric", 
+        month: "short", 
+        day: "numeric",
+        hour: "numeric", 
+        minute: "numeric", 
+        hour12: true, 
+        timeZoneName: "short"
+      }).format(d);
+      
+      el.textContent = formatted;
+    } catch (e) {
+      console.error("Error formatting date:", e);
+    }
+  });
+}
+window.formatDates = formatDates;
+
+function initializeAll() {  
+  try { initializeTableSorting(); } catch (e) { console.warn("initializeTableSorting error:", e); }
+  try { initializeSidebarMenu(); } catch (e) { console.warn("initializeSidebarMenu error:", e); }
+  try { initializeDropdowns(); } catch (e) { console.warn("initializeDropdowns error:", e); }
+  formatDates();
 }
 
 document.addEventListener("DOMContentLoaded", initializeAll);
-document.body.addEventListener("htmx:afterSwap", initializeAll);
+document.body.addEventListener("htmx:afterSwap", function() {
```

Comment:
I see we removed `document.body.addEventListener("htmx:afterSwap", initializeAll);`, is that ok? I see that formatDates is called inside initializeAll. Can you clarify a bit what this change is for?

Reply:
@SonaliBedge thanks! I'm still curious what the setTimeout is for? To me, it seems the original code would still work? 

`document.body.addEventListener("htmx:afterSwap", initializeAll);`